libs/gundeck-types/src/Gundeck/Types/BulkPush.hs

```diff
@@ -0,0 +1,40 @@
+{-# LANGUAGE DeriveGeneric              #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+
+-- | This module defines the types used by the Cannon API.  It is contained in package gundeck-types
+-- for the pragmatic reason that it allows us to re-use types from the gundeck API.  (This move can
+-- be justified by picturing Cannon as a local service that makes only sense in the context of a
+-- Gundeck.)
+module Gundeck.Types.BulkPush where
+
+import Data.Aeson
+import Data.Id
+import GHC.Generics
+import Gundeck.Types.Notification
+
+
+type PushTarget = (UserId, ConnId)
```

Comment:
encoding pairs is a bit odd in JSON, (i.e., it turns `(a, b)` to `[a, b]`)) which is an odd encoding so it'd be better IMHO to have a more explicit structure even if more verbose since this relies on the order of arguments.

Reply:
so, newtype?  will do!

Reply:
8be98a7e25ef94ef6188b0eede7681bec7b03216

Reply:
d05faa95 after rebase